File: PASSWORD-VALIDATOR/PasswordValidator.py
# password Validator
# password must be > greater than 8 characters in length
# must contain atleast 1 lowercase
# atleast on uppercase
# atleast on digit
# and atleast one special character excluding -, will not be counted as special character
# password must not have - and spaces

def password_validator(string):
    lower = 0
    upper = 0
    num = 0
    special = 0
    exclusion = 0

    length = len(string)

    for char in string:
        if char.isupper():
            upper += 1
        elif char.islower():
            lower += 1
        elif char.isdigit():
            num += 1
        elif not char.islower() and not char.isupper() and not char.isdigit():
            special += 1
        elif char.equal('-') or char ==" ":
            exclusion += 1


    print("this is lower", lower)
    print("this is upper", upper)
    print("this is num", num)
    print("this is special", special)
    print("this is exclustion", exclusion)



    if upper > 0 and lower > 0 and num > 0 and special > 0 and length > 8 and exclusion < 1:
        return True
    else:
        return False
    
print(password_validator("SYL5@-viatn"))
    
    

File: PASSWORD-VALIDATOR/passwordValidator.java
public class PasswordValidator{
    public static void main(String[] args){

        //call the method
        String result = isValidPassword("mynaSG$12&");
        System.out.println(result);

    }

    static String isValidPassword(String password){
        // counters
        int upper = 0;
        int lower = 0;
        int num = 0;
        int spec = 0;
    

        //1 the length of the password must be greater than 8
        int size = password.length();


        // go through every character and add to counters
        for (int x = 0; x < size; x++){
            char i = password.charAt(x);

            if(Character.isUpperCase(i)){
                upper += 1;

            }else if (Character.isLowerCase(i)){
                lower += 1;

            }else if(Character.isDigit(i)){
                num += 1;

            }else if ((!Character.isLetter(i))&&(!Character.isDigit(i))){
                spec += 1; 
            }
        }

        // now i use an if statement to determine whether is the password contains atleast on of (lower letter, uppercase letter, number, special character and length > 8)
        if ((upper > 0) && (lower > 0) && (num > 0) && (spec > 0) && (size >= 8)){
            return "password is valid";
            
        }else {
            return "invalid password";
        
        }

    }   
}

/*
NOTES
1. when we want to see the result or how the function behave when we call it in the main class -- 
    ensure to create a variable that holds the results of the class - and print this variable

2. in py we can use not in [ list ], to exclude characters from this list in JAVA
    (!nameoflist.of(items of the list).contains(pass in character))

*/ 

